livekit-agents/livekit/agents/utils/transcription_manager.py

```diff
@@ -0,0 +1,65 @@
+import asyncio
+import time
+from typing import List
+from uuid import uuid4
+
+from livekit import rtc
+
+
+class TranscriptionManager:
+    def __init__(self, room: rtc.Room):
+        self._segments: List[rtc.TranscriptionSegment] = []
+        self._room = room
+
+    def start_segment(
+        self, participant: str | rtc.Participant, language: str, track_id: str
+    ) -> "TranscriptionManagerSegmentHandle":
+        return TranscriptionManagerSegmentHandle(
+            room=self._room,
+            participant=participant,
+            track_id=track_id,
+            language=language,
+        )
+
+
+class TranscriptionManagerSegmentHandle:
+    def __init__(
+        self,
+        room: rtc.Room,
+        participant: str | rtc.Participant,
+        track_id: str,
+        language: str,
+    ):
+        self._id = str(uuid4())
+        self._room = room
+        self._language = language
+        self._track_id = track_id
+        if isinstance(participant, rtc.Participant):
+            self._participant_identity = participant.identity
+        else:
+            self._participant_identity = participant
+        self._start_time = time.time_ns()
+        self._text = ""
+
+    def update(self, text: str):
```

Comment:
hmm shouldn't update publish as well, otherwise it wouldn't be real-time updates?

I think it might be better to have final as a flag on `update`, versus final vs commit. If we handled the STTEvent, then it could be handled automatically.

the other thing is thinking about speed of updates.. I'm wondering if the updates should be pushed into a background queue, so we can do some sort of automatically throttling.. for example, if I did:

```
segment.update('a')
segment.update('a b')
```

this should only trigger a single update. (we can update this in a future PR, but just thinking out loud)